core/commands/resolve.go

```diff
@@ -62,27 +70,19 @@ Resolve te value of another name:
 			}
 		}
 
-		var name string
-
-		if len(req.Arguments()) == 0 {
-			if n.Identity == "" {
-				res.SetError(errors.New("Identity not loaded!"), cmds.ErrNormal)
-				return
-			}
-			name = n.Identity.Pretty()
-
-		} else {
-			name = req.Arguments()[0]
+		name := req.Arguments()[0]
+		recursive, _, _ := req.Option("recursive").Bool()
+		depth := 1
+		if recursive {
+			depth = 0
 		}
 
-		output, err := n.Namesys.Resolve(n.Context(), name)
+		output, err := n.Namesys.Resolve(n.Context(), name, depth)
 		if err != nil {
 			res.SetError(err, cmds.ErrNormal)
 			return
 		}
 
-		// TODO: better errors (in the case of not finding the name, we get "failed to find any peer in table")
```

Comment:
Hmm-- we probably should make them the same.

This is odd:
```
> ipfs /ipfs/QmNonExistentKey
Error: context deadline exceeded
```
cmdslib should just error out...